src/common/ceph_context.cc

```diff
@@ -569,6 +578,17 @@ PerfCountersCollection *CephContext::get_perfcounters_collection()
   return _perf_counters_collection;
 }
 
+void CephContext::enable_perf_counter()
+{
+  ceph_spin_lock(&_cct_perf_lock);
```

Comment:
Hi @liewegas , after reconsideration, I think that lock is still needed, although there is very less chance of contention since set/get methods are most likely one time thing, but for the API completeness and for test purpose, it makes it a little bit easier..

As spinlock rather than Mutex, two reasons: 1) that file does not include "common/Mutex.h", and the vas versa is true.. 2) all locks in that classes are spinlock (properly because of 1)?).  As there is little chance of contention, that might be ok?

Please let me know your concerns. Thanks.